File: archive/find-the-difference-of-two-arrays.js
const findDifference = function(nums1, nums2) {
  let mem1 = []
  let mem2 = []

  for (let i = 0; i < nums1.length; i++) {
    if (nums2.includes(nums1[i])) {
      continue
    } else {
      if (!mem1.includes(nums1[i])) {
        mem1.push(nums1[i])
      }
    }
  }

  for (let i = 0; i < nums2.length; i++) {
    if (nums1.includes(nums2[i])) {
      continue
    } else {
      if (!mem2.includes(nums2[i])) {
        mem2.push(nums2[i])
      }
    }
  }
  return [mem1, mem2]
};

console.log(findDifference([1,2,3, 3], [2,4,6]))

File: go-archive/find-the-difference-of-two-arrays.go
func findDifference(nums1 []int, nums2 []int) [][]int {
	mem1 := map[int]bool{}
	mem2 := map[int]bool{}

	for _, v := range nums1 {
		mem1[v] = true
	}

	for _, v := range nums2 {
		mem2[v] = true
	}

	ans1 := []int{}
	ans2 := []int{}

	for k, _ := range mem1 {
		if !mem2[k] {
			ans1 = append(ans1, k)
		}
	}

	for k, _ := range mem2 {
		if !mem1[k] {
			ans2 = append(ans2, k)
		}
	}

	return [][]int{ans1, ans2}
}
